python/lsst/ts/scheduler/scheduler_csc.py

```diff
@@ -817,6 +888,103 @@ async def put_on_queue(self, targets: list[DriverTarget]) -> None:
                 observing_block=observing_block,
             )
 
+    async def _queue_block_scripts(
+        self, observing_block: ObservingBlock
+    ) -> typing.Generator[int, None, None]:
+        """Queue block scripts in the ScriptQueue.
+
+        Parameters
+        ----------
+        observing_block : `ObservingBlock`
+            Observing block to queue scripts from.
+
+        Yields
+        ------
+        sal_index : `int`
+            A sequence of SAL indices of Scripts.
+
+        Notes
+        -----
+        This method will use `_queue_one_script` to queue the scripts by
+        creating a list of background tasks and waiting for them to finish.
+        The background tasks will acquire a semaphore that will limit the
+        number of consecutive scripts running in the ScriptQueue. If the limit
+        is not met, the method will return soon after all scripts are
+        scheduled. If the limit is reached, the method will block and only
+        make progress once the scripts finish executing.
+
+        If a script execution fail, this raises an exception that is then
+        propagated to this method call, and will adding interrupt further
+        scripts.
```

Comment:
It actually turns out that the exception is raised in `_queue_one_script` and not in this method. Maybe move this part of the doc string to that method instead?